support-frontend/assets/pages/digital-subscription-landing/digitalSubscriptionLanding.jsx

```diff
@@ -69,56 +68,119 @@ const CountrySwitcherHeader = headerWithCountrySwitcherContainer({
   ],
 });
 
-const { optimizeExperiments } = store.getState().common;
+const mapStateToProps = (state) => {
+  const { optimizeExperiments } = state.common;
+  const dailyEditionsExperimentId = 'ZVlxqx3wRDGSDw-HYZLiAQ';
+  const dailyEditionsVariant2 = optimizeExperiments
+    .filter(exp => exp.id === dailyEditionsExperimentId && exp.variant === '1').length !== 0;
+
+  return {
+    dailyEditionsVariant: dailyEditionsVariant2 != null ? dailyEditionsVariant2 : false,
+    optimizeHasLoaded: optimizeExperiments.length > 0,
+  };
+};
+
+type Props = {
+  dailyEditionsVariant: boolean,
+  optimizeHasLoaded: boolean,
+}
 
-const dailyEditionsExperimentId = 'zFSCLqNKT4yQKFNEraigAQ';
-const dailyEditionsVariant = optimizeExperiments
-  .filter(exp => exp.id === dailyEditionsExperimentId && exp.variant === '1').length !== 0;
+type State = {
+  showPage: boolean,
+  numOfChecks: number,
+}
 
 // ----- Render ----- //
+class LandingPage extends Component<Props, State> {
+  state = {
+    showPage: this.props.optimizeHasLoaded,
+    numOfChecks: 0,
+  }
+
+  wait = () => {
+    const { numOfChecks } = this.state;
+    const { optimizeHasLoaded } = this.props;
+
+    if (optimizeHasLoaded || numOfChecks > 16) {
```

Comment:
Is this now just to deal with the case where optimize never loads within the other timeout? I assume the case where optimize does eventually load is handled just by updating the state.

If that's right then can't we just update another bit of state (`showPage`) from `optimize.js` in the path where the timeout is reached?

Reply:
once the component is initialised, state can only be updated by setState (as far as I know) so I don't think we can update it from optimize.js, also I think it's better to have presentation logic controlled by the component